# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/debug/bug1586762.js
# Commit: 8c9daa76d87a
# Full Hash: 8c9daa76d87aef8bf071bef81d9990ccd211983b
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-10-17 21:54:25
# Description:
#   Bug 1586762 - Set BaselineFrame::interpreterICEntry correctly for BaselineDebugModeOSR at debug prologue. r=iain
#   
#   We were using the first pc, but that skips the prologue's type monitor ICs.
#   
#   This bug is pretty hard to trigger in practice (I was unable to write a test
# ==============================================================================

diff -r e641756cd54f -r 8c9daa76d87a js/src/jit-test/tests/debug/bug1586762.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/debug/bug1586762.js	Thu Oct 17 15:33:48 2019 +0000
@@ -0,0 +1,9 @@
+setJitCompilerOption("baseline.warmup.trigger", 0);
+var g = newGlobal({newCompartment: true});
+var dbg = new Debugger;
+dbg.addDebuggee(g);
+g.eval("" + function f() { return 7; });
+dbg.onEnterFrame = function() {
+  dbg.removeDebuggee(g);
+}
+assertEq(g.f(), 7);